08-Check-ODAP-Tables.py

```diff
@@ -0,0 +1,143 @@
+        length(characters) - length(replace(characters, ",", "")) as chr_num,
+        length(fandoms) - length(replace(fandoms, ",", "")) as fnd_num,
+        length(tags) - length(replace(tags, ",", "")) as tag_num
+        FROM
+        stories""",
+    )
+
+    for story in tag_counts:
+        # We're counting commas, not items, so we need to add 4 to the total to
+        # account for the worst-case of each type having one entry.
+        total = (
+            story["cat_num"]
+            + story["chr_num"]
+            + story["fnd_num"]
+            + story["tag_num"]
+            + 4
+        )
+        if total > 75:
+            log.error(f"Found story {story['id']} with too many tags!")
+            found_error = True
+
+    if found_error:
+        log.error("Found error in step O; ending audit here.")
+        sys.exit(1)
+
+    ##
+    ## Step P: Check for excessively long summaries
+    ##
+
+    # The archive does not support summaries longer than 1250 characters.
+
+    log.debug("Starting step P.")
+    found_error = False
+
+    long_sums = sql.execute_dict(
+        "SELECT id, char_length(summary) as len FROM stories HAVING len >= 1250"
+    )
+    if long_sums:
+        found_error = True
+        for story in long_sums:
+            log.error(f"Found story {story['id']} with too long summary!")
+
+    if found_error:
+        log.error("Found error in step P; ending audit here.")
+        sys.exit(2)
+
+    ##
+    ## Step Q: Check for excessively long notes
+    ##
+
+    # The archive does not support notes longer than 5000 characters.
+
+    log.debug("Starting step Q.")
+    found_error = False
+
+    long_notes = sql.execute_dict(
+        "SELECT id, char_length(notes) as len FROM stories HAVING len >= 5000"
+    )
+    if long_notes:
+        found_error = True
+        for story in long_notes:
+            log.error(f"Found story {story['id']} with too long notes!")
+
+    if found_error:
+        log.error("Found error in step Q; ending audit here.")
+        sys.exit(3)
+
+    ##
+    ## Step R: Check for too-long chapters
+    ##
+
+    # Chapters longer than 500,000 characters should be split.
+    # NB: This is counting bytes, not characters, for performance reasons.
+    log.debug("Starting step R.")
+    found_error = False
+
+    long_chap = sql.execute_dict(
+        "SELECT id as chap, story_id as sid, length(text) as len FROM chapters HAVING len > 500000",
+    )
+    if long_chap:
+        found_error = True
+        for story in long_chap:
+            log.error(
+                f"Found chapter {story['chap']} in story {story['sid']} that could be too long.  Check to make sure it's under 500k characters."
+            )
+
+    if found_error:
+        log.error("Found error in step R; ending audit here.")
+        sys.exit(4)
+
+    ##
+    ## Step S: Check for stories with too many chapters
+    ##
+
+    # The archive does not permit stories to have more than 200 chapters.
```

Comment:
```suggestion
    # The Archive does not allow importing more than 200 chapters per story.
```
(just to be pedantic on why we have this arbitrary restriction (see [Archive config here](https://github.com/otwcode/otwarchive/blob/d87652399df73fa5e3b829b0089529501a381538/config/config.yml#L158))